include/bluetooth/mesh/gen_prop_srv.h

```diff
@@ -0,0 +1,243 @@
+ * @defgroup bt_mesh_prop_srv Generic Property server models
+ * @{
+ * @brief API for the Generic Property server models.
+ */
+#ifndef BT_MESH_GEN_PROP_SRV_H__
+#define BT_MESH_GEN_PROP_SRV_H__
+
+#include <bluetooth/mesh/gen_prop.h>
+#include <bluetooth/mesh/model_types.h>
+#include <sys/util.h>
+
+struct bt_mesh_prop_srv;
+
+/** @def BT_MESH_PROP_SRV_INIT
+ *
+ * @brief Initialization parametersr for a @ref bt_mesh_prop_srv instance.
+ *
+ * @param[in] _properties Array of properties supported by the server.
+ * @param[in] _property_count Number of properties supported by the server.
+ * @param[in] _get Getter handler for property values. @sa
+ * bt_mesh_prop_srv::get.
+ * @param[in] _set Setter handler for property values. @sa
+ * bt_mesh_prop_srv::set.
+ */
+#define BT_MESH_PROP_SRV_INIT(_properties, _property_count, _get, _set)        \
+	{                                                                      \
+		.get = _get, .set = _set, .properties = _properties,           \
+		.property_count = _property_count,                             \
+		.pub = {                                                       \
+			.update = _bt_mesh_prop_srv_update_handler,            \
+			.msg = NET_BUF_SIMPLE(MAX(                             \
+				BT_MESH_MODEL_BUF_LEN(                         \
+					BT_MESH_PROP_OP_MFR_PROP_STATUS,       \
+					BT_MESH_PROP_MSG_MAXLEN_PROP_STATUS),  \
+				BT_MESH_MODEL_BUF_LEN(                         \
+					BT_MESH_PROP_OP_MFR_PROPS_STATUS,      \
+					2 * _property_count))),                \
+		},                                                             \
+	}
+
+/** @def BT_MESH_PROP_SRV_ADMIN_INIT
+ *
+ * @brief Initialization parameters for a @ref bt_mesh_prop_srv acting as a
+ * Generic Admin Property Server.
+ *
+ * @param[in] _properties Array of properties exposed by the server.
+ * @param[in] _get Getter function for property values.
+ * @param[in] _set Setter function for property values.
+ */
+#define BT_MESH_PROP_SRV_ADMIN_INIT(_properties, _get, _set)                   \
+	BT_MESH_PROP_SRV_INIT(_properties, ARRAY_SIZE(_properties), _get, _set)
+
+/** @def BT_MESH_PROP_SRV_MFR_INIT
+ *
+ * @brief Initialization parameters for a @ref bt_mesh_prop_srv acting as a
+ * Generic Manufacturer Property Server.
+ *
+ * @param[in] _properties Array of properties exposed by the server.
+ * @param[in] _get Getter function for property values.
+ */
+#define BT_MESH_PROP_SRV_MFR_INIT(_properties, _get)                           \
+	BT_MESH_PROP_SRV_INIT(_properties, ARRAY_SIZE(_properties), _get, NULL)
+
+/** @def BT_MESH_PROP_SRV_CLIENT_INIT
+ *
+ * @brief Initialization parameters for a @ref bt_mesh_prop_srv acting as a
+ * Generic Client Property Server.
+ *
+ * @param[in] _properties Array of properties exposed by the server. May be
+ * constant.
+ */
+#define BT_MESH_PROP_SRV_CLIENT_INIT(_properties)                              \
+	BT_MESH_PROP_SRV_INIT((struct bt_mesh_prop *)_properties,              \
```

Comment:
Is it possible to use `struct bt_mesh_prop[]` here?

Reply:
Yes